src/vellum/lib/test_suites/resources.py

```diff
@@ -0,0 +1,134 @@
+import os
+import time
+from typing import Callable
+
+from src.vellum.lib.test_suites.constants import DEFAULT_MAX_POLLING_DURATION_MS, DEFAULT_POLLING_INTERVAL_MS
+from src.vellum.lib.test_suites.exceptions import TestSuiteRunResultsException
+from src.vellum.types.named_test_case_variable_value_request import NamedTestCaseVariableValueRequest
+from src.vellum.types.test_case_variable_value import TestCaseVariableValue
+from src.vellum.client import Vellum
+from src.vellum.types import (
+    TestSuiteRunExecution,
+    TestSuiteRunMetricOutput,
+    TestSuiteRunState,
+    TestSuiteRunExternalExecConfigRequest,
+    TestSuiteRunExternalExecConfigDataRequest,
+    ExternalTestCaseExecution,
+)
+
+
+class VellumTestSuiteResults:
+    def __init__(
+        self,
+        id: str,
+        polling_inteval: int = DEFAULT_POLLING_INTERVAL_MS,
+        max_polling_duration: int = DEFAULT_MAX_POLLING_DURATION_MS,
+    ) -> None:
+        self.id = id
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self._state = TestSuiteRunState.QUEUED
+        self._executions: list[TestSuiteRunExecution] | None = None
+        self._polling_interval = polling_inteval
+        self._max_polling_duration = max_polling_duration
+
+    def get_metric_outputs(
+        self, metric_identifier: str | None = None, output_identifier: str | None = None
+    ) -> list[TestSuiteRunMetricOutput]:
+        executions = self._get_test_suite_run_executions()
+
+        all_metric_results = [execution.metric_results for execution in executions]
+
+        filtered_metric_outputs = [
+            metric_result.outputs
+            for metric_result_execution in all_metric_results
+            for metric_result in metric_result_execution
+            if not metric_identifier
+            or (metric_result.metric_id == metric_identifier)
+            or (metric_result.metric_label == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.id == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.name == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.label == metric_identifier)
+        ]
+
+        filtered_metric_filtered_outputs = [
+            metric_output
+            for metric_output_execution in filtered_metric_outputs
+            for metric_output in metric_output_execution
+            if not output_identifier or metric_output.name == output_identifier
+        ]
+
+        return filtered_metric_filtered_outputs
+
+    def _refresh_test_suite_run(self):
```

Comment:
nit: The current naming made me think we were going to re-run the TSR

```suggestion
    def _refresh_test_suite_run_state(self):
```